hummingbot/connector/exchange/ascend_ex/ascend_ex_api_order_book_data_source.py

```diff
@@ -33,28 +35,41 @@ def logger(cls) -> HummingbotLogger:
             cls._logger = logging.getLogger(__name__)
         return cls._logger
 
-    def __init__(self, trading_pairs: List[str] = None):
+    def __init__(self, throttler: AsyncThrottler, trading_pairs: List[str] = None):
         super().__init__(trading_pairs)
+        self._throttler = throttler
         self._trading_pairs: List[str] = trading_pairs
         self._snapshot_msg: Dict[str, any] = {}
 
     @classmethod
-    async def get_last_traded_prices(cls, trading_pairs: List[str]) -> Dict[str, float]:
+    def _get_throttler_instance(cls) -> AsyncThrottler:
+        throttler = AsyncThrottler(CONSTANTS.RATE_LIMITS)
+        return throttler
+
+    @classmethod
+    async def get_last_traded_prices(
+        cls, trading_pairs: List[str], throttler: Optional[AsyncThrottler] = None
+    ) -> Dict[str, float]:
         result = {}
 
         for trading_pair in trading_pairs:
             async with aiohttp.ClientSession() as client:
-                resp = await client.get(f"{REST_URL}/trades?symbol={convert_to_exchange_trading_pair(trading_pair)}")
+                throttler = throttler or cls._get_throttler_instance()
```

Comment:
Similar to the comment left in the NDAX PR. I think assigning a default value would be cleaner.

Reply:
Ditto on mutable default variables and single-instantiation point